### ISSUE TITLE
Keep git repositories checked out

### ISSUE THREAD

**Issue by dolfandringa** [python-poetry/poetry#3414]
This may be a feature request, but didn't want to open one without knowing for sure it's not possible already. 

TLDR: is it possible to keep VCS dependencies checked out in a `src` folder insgtead of requiring it to be checked out for every `poetry update` or `poetry install`?

So I have a package that depends on a pacakage from another git repository. That git repository is 1.8Gb in size. On top of that I am dealing with rate limiting from github which often limits my git checkouts to 10KB/s (don't know why). That last part is solved with a VPN, but even without the rate limiting, it would really save a lot of time when reinstalling dependencies multiple times if there was an option to keep VCS dependencies checked out in some `src` folder and just git pull/fetching updates, instead of checking out the whole repository again and again to `/tmp`. Is this possible somehow?

- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

**Comment by sinoroc** [comment]
Makes somewhat sense. Don't know what the current behaviour (internally) exactly is, though. Have you tried setting `develop = true` on those _git_ dependencies? Also maybe there are improvement that could be made in _poetry_ so that only a single commit is fetched (or at least not the whole repository, although maybe poetry already fetches selectively, I don't know).

**Comment by dolfandringa** [comment]
When I set develop=true on that repository and then run a `poetry update` it still runs this git command:
```git clone --recurse-submodules http://github.com/dolfandringa/KivyMD.git /tmp/pypoetry-git-KivyMDivv85z7i```so the full checkout in /tmp. And every subsequent run I guess will do the same again.

**Comment by dolfandringa** [comment]
I made a feature request for this: https://github.com/python-poetry/poetry/issues/3420
